### PR TITLE
statefulSet: detailed revision view

### LINKED ISSUE #92292: StatefulSet: detailed revision view return a revisions list
**What happened**:
build kubectl
```
[user@localhost kubernetes-master]# make kubectl
```
when I view the statefulset with a revision, it return a list of the statefulset revisions
```
[user@localhost kubernetes-master]# _output/bin/kubectl rollout history statefulset xxx --revision=1
statefulset.apps/xxx with revision #1
REVISION
1
```

**What you expected to happen**:
when I view the statefulset with a revision, it return the pod template just like deployment and daemonset
```
[user@localhost kubernetes-master]#  _output/bin/kubectl rollout history statefulset xxx --revision=1
statefulset.apps/xxx with revision #1
Pod Template:
  Labels:
        app=xxx
        name=xxx
        version=1.0.0
  Containers:
   rabbitmq:
    Image:      xxxx
    Ports:      8080/TCP
    Host Ports: 0/TCP, 0/TCP
```
**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): master
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by nfyxhan**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #92292

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by nfyxhan** [comment]
I signed it

**Comment by nfyxhan** [comment]
I signed it

File: staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go [hunk 1]
```diff
@@ -233,26 +233,50 @@ type StatefulSetHistoryViewer struct {
...
-	for _, revision := range history {
-		revisions = append(revisions, revision.Revision)
+
+	// Print details of a specific revision
+	if revision > 0 {
+		history, ok := historyInfo[revision]
+		if !ok {
+			return "", fmt.Errorf("unable to find the specified revision")
+		}
+		ssOfHistory, err := applyStatefulSetHistory(ss, history)
```
Comment (whalecold) [inline staging/src/k8s.io/kubectl/pkg/polymorphichelpers/history.go:256]: a similar function has exist `applyRevision`, there is no need to  duplicate it.

**Pull request closed without merging**